src/lib.rs

```diff
@@ -137,6 +132,11 @@ pub fn make_cargo_build_command(
         .stdin(Stdio::inherit())
         .stderr(Stdio::inherit());
 
+    if !sysroot.join("lib/rustlib/armv6k-nintendo-3ds").exists() {
+        eprintln!("No pre-build std found, using build-std");
+        command.arg("-Z").arg("build-std");
+    }
```

Comment:
My only worry with moving this all the way here is that if the user passed any `--` args, this will be included as part of that, instead of passed to `cargo` like we want. Maybe we can just move it after `message_format` but before `args` in the command builder? Could also be something like this (or any alternative)

```rust
    let build_std = if sysroot.join("lib/rustlib/armv6k-nintendo-3ds").exists() {
	    &[][..]
	} else {
	    eprintln!("No pre-build std found, using build-std");
	    &["-Z", "build-std"][..]
	};
```

Reply:
Yeah that's a good point.

Reply:
Fixed